src/ui/public/state_management/state_monitor.js

```diff
@@ -8,9 +8,14 @@ function stateMonitor(state, defaultState) {
   let destroyed = false;
   let ignoredProps = [];
   let changeHandlers = [];
+  let originalState;
 
-  // state.toJSON returns a reference, clone so we can mutate it safely
-  const originalState = cloneDeep(defaultState) || cloneDeep(state.toJSON());
+  setOriginalState(defaultState);
+
+  function setOriginalState(defaultState) {
```

Comment:
Renaming `defaultState` to `initialState` sounds good to me. And then I think you can continue to make things more consistent above:

```javascript
function stateMonitor(state, customInitialState) {
```

(We'll have to name this something other than `initialState` to avoid a collision)